### PR TITLE
Fix bug in populating hmm mean and covariance.

### PR DESCRIPTION
**Pull request by mattbrown11**
Fixed a bug in how the HMM's mean and covariance are populated in the  ActivityHMM.__init__, which yields much better results. Added a `load_and_discretize_data` function to efficiently read group truth and detection json.

### CODE REVIEW

**Comment by mattbrown11** [comment]
@Purg @joshanderson-kw PTAL. simulate_data.py is my sandboxy script, so I wouldn't spend too much effort reviewing it.

**Review by joshanderson-kw — APPROVED** [review]
LGTM, I didn't look at the simulate_data script or the helper functions though.

File: angel_system/activity_hmm/core.py [hunk 1]
```diff
@@ -743,11 +773,219 @@ def score_raw_detections(X, Z, plot_results=False):
...
+            'start': time_from_name(row["start_frame"]),
+            #'end': float(RE_FILENAME_TIME.match(row["end_frame"]).groups()[0])
+            'end': time_from_name(row["end_frame"])
+        }
+        gt.append(g)
+
+    print(f"Loaded ground truth from {activity_gt}")
+    gt = pd.DataFrame(gt)
+
+    detections_input = [det for det in (literal_eval(s) for s in open(extracted_activity_detections))][0]
```
Comment (Purg) [inline angel_system/activity_hmm/core.py:859]: `json.load` please, not AST. If the source file is not formatted correctly for `json.load` to operate, that is a bug and it should be fixed.

File: angel_system/activity_hmm/core.py [hunk 2]
```diff
@@ -275,37 +279,63 @@ def __init__(self, dt, class_str, med_class_duration,
...
 
-            class_conf_mean_mat = np.zeros((N_, N_))
-            sphere_conf_cov_mat = np.zeros(N_)
-
-            # Square to turn from std to cov. We'll reuse the name.
-            class_std_conf = class_std_conf**2
+            if np.ndim(class_mean_conf) == 1:
+                class_mean_conf = np.diag(class_mean_conf)
+            elif np.ndim(class_mean_conf) > 2:
+                raise Exception('np.ndim(class_mean_conf) must be 1 or 2')
```
Comment (Purg) [inline angel_system/activity_hmm/core.py:300]: Generally do not use base-class exceptions, [concrete exceptions](https://docs.python.org/3/library/exceptions.html#concrete-exceptions) should be used if looking for a builtin, define your own exception, or use an assert.

File: angel_system/activity_hmm/core.py [hunk 3]
```diff
@@ -743,11 +773,219 @@ def score_raw_detections(X, Z, plot_results=False):
...
+
     return auc
+
+
+def load_and_discretize_data(activity_gt: str,
+                             extracted_activity_detections: str,
+                             time_window: float, uncertain_pad: float):
+    """
+
+    Parameters
```
Comment (Purg) [inline angel_system/activity_hmm/core.py:819]: Missing description about what this is supposed to do.

File: angel_system/activity_hmm/core.py [hunk 4]
```diff
@@ -743,11 +773,219 @@ def score_raw_detections(X, Z, plot_results=False):
...
+    -------
+    TYPE
+        DESCRIPTION.
+
+    """
+    gt_f = pd.read_feather(activity_gt)
+    # Keys: class, start_frame,  end_frame, exploded_ros_bag_path
+
+    gt = []
+    RE_FILENAME_TIME = re.compile(r"frame_\d+_(\d+)_\d+.\w+")
```
Comment (Purg) [inline angel_system/activity_hmm/core.py:845]: This is no longer used, I only see it in commented-out code, which could also be removed.

Comment (mattbrown11) [reply angel_system/activity_hmm/core.py]: Addressed.

File: angel_system/activity_hmm/core.py (hunk 3 above)
Comment (mattbrown11) [reply angel_system/activity_hmm/core.py]: Addressed.

File: angel_system/activity_hmm/core.py (hunk 2 above)
Comment (mattbrown11) [reply angel_system/activity_hmm/core.py]: Addressed.

File: angel_system/activity_hmm/core.py (hunk 1 above)
Comment (mattbrown11) [reply angel_system/activity_hmm/core.py]: Addressed.

**Review by Purg — APPROVED** [review]

**Pull request merged**